lib/Sema/CSApply.cpp

```diff
@@ -264,14 +264,12 @@ static Expr *buildDynamicMemberLookupIndexExpr(StringRef name, SourceLoc loc,
                                                ConstraintSystem &cs) {
   auto &ctx = cs.TC.Context;
 
-  auto *stringDecl = ctx.getStringDecl();
-  auto stringType = stringDecl->getDeclaredType();
-
   // Build and type check the string literal index value to the specific
   // string type expected by the subscript.
   auto *nameExpr = new (ctx) StringLiteralExpr(name, loc, /*implicit*/true);
-  nameExpr->setBuiltinInitializer(ctx.getStringBuiltinInitDecl(stringDecl));
-  nameExpr->setType(stringType);
+  nameExpr->setBuiltinInitializer(ctx.getStringBuiltinInitDecl(
+                                    ctx.getStringDecl()));
+  nameExpr->setType(ctx.getStringType());
```

Comment:
I don't think this ended up cleaner. It should still be okay to say things the long way.

Reply:
In this specific example, or across the board?

Reply:
In this specific example. None of the others jumped out at me like this where repeated use seemed less clear, though I do hope it doesn't have any noticeable effect on compile time.